Src/WitsmlExplorer.Frontend/components/Modals/LogComparisonModal.tsx

```diff
@@ -0,0 +1,231 @@
+        const fetchSource = LogObjectService.getLogCurveInfo(wellUid, wellboreUid, sourceLog.uid);
+        const fetchTarget = LogObjectService.getLogCurveInfoFromServer(wellUid, wellboreUid, sourceLog.uid, targetCredentials);
+        return {
+          sourceLogCurveInfo: await fetchSource,
+          targetLogCurveInfo: await fetchTarget
+        };
+      };
+      const { sourceLogCurveInfo, targetLogCurveInfo } = await fetchCurves();
+      if (sourceLogCurveInfo.length == 0) {
+        dispatchOperation({ type: OperationType.HideModal });
+        const failureMessageSource = "Unable to compare the log as no log curve infos could be fetched from the source log.";
+        displayMissingLogModal(sourceServer, wellUid, wellboreUid, sourceLog.uid, dispatchOperation, failureMessageSource);
+        return;
+      } else if (targetLogCurveInfo.length == 0) {
+        dispatchOperation({ type: OperationType.HideModal });
+        const failureMessageTarget = "Unable to compare the log as either the log does not exist on the target server or the target log is empty.";
+        displayMissingLogModal(targetServer, wellUid, wellboreUid, sourceLog.uid, dispatchOperation, failureMessageTarget);
+        return;
+      } else {
+        setSourceLogCurveInfo(sourceLogCurveInfo);
+        setTargetLogCurveInfo(targetLogCurveInfo);
+      }
+    };
+    setCurves();
+  }, []);
+
+  useEffect(() => {
+    if (indexesToShow !== null || sourceLogCurveInfo === null || targetLogCurveInfo === null) {
+      return;
+    }
+    const sourceType = sourceLogCurveInfo[0].minDateTimeIndex == null ? "depth" : "time";
+    const targetType = targetLogCurveInfo[0].minDateTimeIndex == null ? "depth" : "time";
+    const indexTypesMatch = sourceType == targetType;
+
+    const mismatchedIndexes = [];
+    if (indexTypesMatch) {
+      for (const sourceCurve of sourceLogCurveInfo) {
+        const targetCurve = targetLogCurveInfo.find((targetCurve) => targetCurve.mnemonic == sourceCurve.mnemonic);
+        if (!targetCurve || areMismatched(sourceCurve, targetCurve)) {
+          mismatchedIndexes.push(logCurveInfoToIndexes(sourceCurve, targetCurve));
+        }
+      }
+      for (const targetCurve of targetLogCurveInfo) {
+        const sourceCurve = sourceLogCurveInfo.find((sourceCurve) => sourceCurve.mnemonic == targetCurve.mnemonic);
+        if (!sourceCurve) {
+          mismatchedIndexes.push(logCurveInfoToIndexes(sourceCurve, targetCurve));
+        }
+      }
+    }
+    setSourceType(sourceType);
+    setTargetType(targetType);
+    setIndexTypesMatch(indexTypesMatch);
+    setIndexesToShow(mismatchedIndexes);
+  }, [sourceLogCurveInfo, targetLogCurveInfo]);
+
+  return (
+    <ModalDialog
+      heading={`Log comparison`}
+      content={
+        <>
+          {(indexesToShow && (
+            <>
+              <span>
+                <p>Source well name: {sourceLog.wellName}</p>
```

Comment:
There should not be any need to specify "`Source` Well", `Wellbore` or `Log`. They are being represented for both source and target

Reply:
_Source_ well, wellbore and log names are _expected_ to match the target, but given that a match is based on UID, it is possible that they differ. An alternative I can see is to show both the source and target names, either always or only when they differ (thus having only one field per name).

Reply:
Ah I see. Yes then it might make sense to have both displayed or at least include the Source tag